app/src/main/java/com/cornellappdev/uplift/ui/screens/ClassDetailScreen.kt

```diff
@@ -0,0 +1,336 @@
+                    .offset(y = 42.dp)
+                    .align(Alignment.BottomCenter)
+                    .graphicsLayer {
+                        alpha = 2f
+                        translationY = -0.5f * scrollState.value.toFloat()
+                    }
+            ) {
+                Box(modifier = Modifier.fillMaxSize()) {
+                    Text(
+                        text = "${upliftClass?.minutes} MIN",
+                        fontWeight = FontWeight(700),
+                        fontSize = 14.sp,
+                        lineHeight = 17.07.sp,
+                        textAlign = TextAlign.Center,
+                        color = PRIMARY_BLACK,
+                        fontFamily = montserratFamily,
+                        modifier = Modifier
+                            .align(Alignment.TopCenter)
+                            .offset(y = 19.dp)
+                    )
+                }
+            }
+        }
+
+        // Date & Time Information
+        Column(
+            modifier = Modifier
+                .fillMaxWidth()
+                .background(Color.White),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            Spacer(Modifier.height(24.dp))
+            Text(
+                text = upliftClass?.dateAsString() ?: "",
+                fontWeight = FontWeight(300),
+                fontSize = 16.sp,
+                lineHeight = 19.5.sp,
+                textAlign = TextAlign.Center,
+                color = PRIMARY_BLACK,
+                fontFamily = montserratFamily
+            )
+            Spacer(Modifier.height(8.dp))
+            Text(
+                text = upliftClass?.time.toString(),
+                fontWeight = FontWeight(500),
+                fontSize = 16.sp,
+                lineHeight = 19.5.sp,
+                textAlign = TextAlign.Center,
+                color = PRIMARY_BLACK,
+                fontFamily = montserratFamily
+            )
+            Spacer(Modifier.height(24.dp))
+            Icon(
+                painter = painterResource(id = R.drawable.ic_calendar),
+                contentDescription = "Add to calendar",
+                tint = PRIMARY_BLACK,
+                modifier = Modifier.clickable {
+                    upliftClass?.let {
+                        val intent = Intent(Intent.ACTION_EDIT)
+                        intent.type = "vnd.android.cursor.item/event"
+                        intent.putExtra("beginTime", it.time.start.timeInMillis(it.date))
+                        intent.putExtra("allDay", false)
+                        intent.putExtra("endTime", it.time.end.timeInMillis(it.date))
+                        intent.putExtra(CalendarContract.Events.EVENT_LOCATION, it.location)
+                        intent.putExtra("title", it.name)
+                        startActivity(context, intent, null)
+                    }
+                }
+            )
+            Spacer(Modifier.height(4.dp))
+            Text(
```

Comment:
it would also be more intuitive plus have a bigger area to click on if the text was also able to make the calendar event.  So please make the text send the calendar intent as well.

Reply:
Good suggestion! Changed in commit [fe4cffa](https://github.com/cuappdev/uplift-android/pull/3/commits/fe4cffa058cfcecbd5266a4db606e7fa50f91284).